File: practicas/practica_2/Deepspace_Ruby_P2/lib/WeaponType.rb
#encoding:utf-8

# File: WeaponType.rb
# Author: Juan Manuel Rodríguez Gómez

module Deepspace

    # Brief: Representa a los tipos de armas del juego
    #        más cierta funcionalidad adicional.
    module WeaponType

        class Type
            
            # Brief: Constructor con un parámetro
            # Param p_power: Poder del arma
            def initialize(p_power)
                @power = p_power;
            end
          
            # Brief: Consultor del poder del arma
            # Return: power
            def power
                @power
            end

            # Brief: Función para representar el enumerado WeaponType
            # Return: String que representa el enumerado WeaponType
            def to_s
                if self == WeaponType::LASER
                  return "LASER"
                elsif self == WeaponType::MISSILE
                  return "MISSILE"
                else
                  return "PLASMA"
                end
              end
        end # Class

        LASER = Type.new(2.0)
        MISSILE = Type.new(3.0)
        PLASMA = Type.new(4.0) 
    end # Module WeaponType
end # Module Deepspace

File: practicas/practica_2/deepspace_Java_P2/src/main/java/deepspace/WeaponType.java
/**
 * @file WeaponType.java
 * @author Juan Manuel Rodríguez Gómez
 */
package deepspace;

/**
 * @brief Representa a los tipos de armas del juego
 *        más cierta funcionalidad adicional.
 */
public enum WeaponType {
    LASER(2.0f),
    MISSILE(3.0f),
    PLASMA(4.0f);
    
    private float power;
    
    /**
     * @brief Constructor con un parámetro
     * @param p_power Poder del arma
     */
    WeaponType(float p_power) {
        power = p_power;
    }
    
    /**
     * @brief Consultor del poder del arma
     * @return power
     */
    final float getPower() {
        return power;
    }
}